src/nvim/ui_client.c

```diff
@@ -283,6 +284,60 @@ void ui_client_event_raw_line(GridLineEvent *g)
                (const schar_T *)grid_line_buf_char, grid_line_buf_attr);
 }
 
+void ui_client_event_connect(Array args)
+{
+  if (args.size < 2 || args.items[0].type != kObjectTypeString || args.items[1].type != kObjectTypeBoolean) {
+    ELOG("Error handling UI event 'connect'");
+    return;
+  }
+
+  char *server_addr = args.items[0].data.string.data;
+  bool stop_server = args.items[1].data.boolean;
+
+  ELOG("stop_server: %d", stop_server);
+  multiqueue_put(main_loop.fast_events, channel_connect_event, server_addr, (void *) stop_server);
+}
+
+static void channel_connect_event(void **argv)
+{
+  char *server_addr = argv[0];
+  bool stop_server = (bool) argv[1];
+
+
+  Error err = ERROR_INIT;
+  nvim_command(cstr_as_string("detach"), &err);
```

Comment:
We're already in the client, and `ex_connect` already set `chan->detach=true`. No need to run `:detach` on the server.

Instead try going straight to `"qall!"` (set `stop_server=true` here, I guess).